### PR TITLE
fix interaction with planar joints

### PR DESCRIPTION
**Pull request by rhaschke**
Rotation angle for planar joint was computed based on Euler angles which exhibits discontinuities.
Better use dedicated method setJointPositions(joint_name, pose).

### CODE REVIEW

**Comment by davetcoleman** [comment]
@mcevoyandy can you review this, i'm not sure how to verify this makes sense

**Comment by rhaschke** [comment]
> I'm not sure how to verify this makes sense.

I use a mobile robot, e.g. PR2, defined a planar virtual base joint, defined a group for this, and started rviz to move the robot around...

File: moveit_ros/robot_interaction/src/interaction_handler.cpp [hunk 1]
```diff
@@ -338,35 +338,13 @@ void InteractionHandler::updateStateEndEffector(robot_state::RobotState* state,
...
-    vals[vj->joint_name + "/trans_x"] = rel_pose.position.x;
-    vals[vj->joint_name + "/trans_y"] = rel_pose.position.y;
-    vals[vj->joint_name + "/trans_z"] = rel_pose.position.z;
-    vals[vj->joint_name + "/rot_x"] = q.x();
-    vals[vj->joint_name + "/rot_y"] = q.y();
-    vals[vj->joint_name + "/rot_z"] = q.z();
-    vals[vj->joint_name + "/rot_w"] = q.w();
-  }
-  state->setVariablePositions(vals);
+  state->setJointPositions(vj->joint_name, pose);
```
Comment (jrgnicho) [inline moveit_ros/robot_interaction/src/interaction_handler.cpp]: I recently wrote some code that updated planar joints and verified that the `setJointPositions` method ensures that only the unconstrained joint DOFs are updated.

**Comment by v4hn** [comment]
O.o
There is no sense to have the logic that was in there in the interaction handler.
I recently worked with the `setJointPositions` method too and it is meant to deal with this exactly.

Thanks for the patch, merging.

**Pull request merged**